examples/workspaces_variables/manager/main.tf

```diff
@@ -0,0 +1,156 @@
+}
+
+variable "vcs_repo_identifier" {
+  description = <<-EOT
+  The format of VCS repo identifier might differ depending on the VCS provider,
+  see https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/workspace
+  EOT
+}
+
+variable "vcs_token" {
+  description = "The VCS token should correspond to an API token that can create OAuth clients."
+}
+
+variable "vars_mapped_by_workspace_name" {
+    description = <<-EOT
+    This is the map of workspaces and variables. A workspace is created for each
+    top level key and then variables are set on the workspace.
+    EOT
+}
+
+variable "additional_vars" {
+  description = "This is a map of additional variables intended to be set in specific workspaces."
+  default = {
+    customer_1_workspace = {
+      i_am_sensitive_tf_var = {
+        value = "i am sensitive"
+        sensitive = true
+      }
+    }
+  }
+}
+
+variable "default_var_category" {
+  description = "Default category for variables being set in managed workspaces unless specified"
+  default = "terraform"
+}
+
+variable "default_var_hcl" {
+  description = "By default, variables being set in managed workspaces will not be interpreted as hcl values"
+  default = false
+}
+
+variable "default_var_sensitive" {
+  description = "By default, variables being set in managed workspaces will be non-sensitive"
+  default = false
+}
+
+locals {
+  # Flatten the workspaces variable nested structure for later iteration in the managed_var resource. Adapted from:
+  # https://developer.hashicorp.com/terraform/language/functions/flatten#flattening-nested-structures-for-for_each
+
+  # Results in a list of maps, that contains all the information
+  # required to manage that workspace variable.
+  #   [{
+  #     ws            = ws_name
+  #     var_key       = name
+  #     var_value     = value
+  #     var_category  = string
+  #     var_hcl       = true/false
+  #     var_sensitive = true/false
+  #     ws_id         = <tfe_workspace>.id
+  #   }...]
+  ws_variables = flatten([
+    for ws_name, variables in var.vars_mapped_by_workspace_name : [
+      for var_name, var_attrs in merge(variables, lookup(var.additional_vars, ws_name, {})) : {
+        ws            = ws_name
+        var_key       = var_name
+        var_value     = var_attrs["value"]
+        var_category  = lookup(var_attrs, "category",  var.default_var_category)
+        var_hcl       = lookup(var_attrs, "hcl",       var.default_var_hcl)
+        var_sensitive = lookup(var_attrs, "sensitive", var.default_var_sensitive)
+        ws_id         = tfe_workspace.managed_ws[ws_name].id
+      }
+    ]
+  ])
+}
+
+# This example oauth connection assumes the VCS provider is Github.
+resource "tfe_oauth_client" "gh" {
+  organization     = var.tf_organization
+  api_url          = "https://api.github.com"
+  http_url         = "https://github.com"
+  oauth_token      = var.vcs_token
```

Comment:
I wonder if the `vcs_token` permissions should be explicitly named in this example? 🤔  When I ran an apply, the first time, I got an `unauthorized`.